src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs

```diff
@@ -265,6 +264,87 @@ public Task EnforceReelection()
             return Task.CompletedTask;
         }
 
+        /* Promote a non-voter to a promotable */         
+        [RavenAction("/admin/cluster/promote", "POST", "/admin/cluster/promote?nodeTag={nodeTag:string}")]
+        public async Task PromoteNode()
+        {
+            if (ServerStore.LeaderTag == null)
+                return;
+            
+            if (ServerStore.IsLeader() == false)
+            {
+                RedirectToLeader();
+                return;
+            }
+
+            var nodeTag = GetStringQueryString("nodeTag");
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var topology = ServerStore.GetClusterTopology(context);
+                if (topology.AllNodes.ContainsKey(nodeTag) == false)
+                {
+                    throw new InvalidOperationException(
+                        $"Failed to promote node {nodeTag} beacuse {nodeTag} is not a part of the cluster topology");
+                }
+                if (topology.Watchers.ContainsKey(nodeTag) == false)
```

Comment:
Why check twice, you can do it once and check if the tag is in the watchers and just error otherwise.